backends/platform/libretro/Makefile

```diff
@@ -586,32 +586,125 @@ else
 endif
 endif
 
+######################################################################
+# Dependencies
+######################################################################
+
+libdeps_libretro_common.a: $(OBJS_DEPS_LIBRETRO_COMMON)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES = libdeps_libretro_common.a
+
+ifeq ($(USE_FLUIDSYNTH), 1)
+libdeps_fluidsynth.a: $(OBJS_DEPS_FLUIDSYNTH)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_fluidsynth.a
+endif
+
+ifeq ($(USE_FLAC), 1)
+libdeps_libflac.a: $(OBJS_DEPS_LIBFLAC)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_libflac.a
+endif
+
+ifeq ($(USE_VORBIS), 1)
+libdeps_libvorbis.a: $(OBJS_DEPS_LIBVORBIS)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_libvorbis.a
+endif
+
+ifeq ($(USE_TREMOR), 1)
+libdeps_tremor.a: $(OBJS_DEPS_TREMOR)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_tremor.a
+endif
+
+ifeq ($(USE_ZLIB), 1)
+libdeps_libz.a: $(OBJS_DEPS_LIBZ)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_libz.a
+endif
+
+ifeq ($(USE_MAD), 1)
+libdeps_libmad.a: $(OBJS_DEPS_LIBMAD)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_libmad.a
+endif
+
+ifeq ($(USE_FAAD), 1)
+libdeps_faad.a: $(OBJS_DEPS_FAAD)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_faad.a
+endif
+
+ifeq ($(USE_PNG), 1)
+libdeps_png.a: $(OBJS_DEPS_PNG)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_png.a
+endif
+
+ifeq ($(USE_JPEG), 1)
+libdeps_jpeg.a: $(OBJS_DEPS_JPEG)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_jpeg.a
+endif
+
+ifeq ($(USE_THEORADEC), 1)
+libdeps_theora.a: $(OBJS_DEPS_THEORA)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_theora.a
+endif
+
+ifeq ($(USE_FREETYPE2), 1)
+libdeps_freetype.a: $(OBJS_DEPS_FREETYPE)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_freetype.a
+endif
+
+ifeq ($(USE_FRIBIDI), 1)
+libdeps_fribidi.a: $(OBJS_DEPS_FRIBIDI)
+	@echo Linking $@...
+	$(HIDE)$(AR) $@ $^
+LIBDEPS_ARCHIVES += libdeps_fribidi.a
```

Comment:
I would move all these blocks in the relevant section in `dependencies.mk` (i.e. [before this line](https://github.com/scummvm/scummvm/blob/13798078336d0149833c962e11ec6a6feb3b3b84/backends/platform/libretro/dependencies.mk#L580) in this case), to keep all dependencies stuff together and segregated in a modularized way.
Also because this if block does not cover the case where `USE_xyz=1` but `USE_SYSTEM_xyz=1` as well, hence `libdeps_xyx.a` is not applicable.